native-src/virtual_drive/convert_to_placeholder.cpp

```
@@ -1,29 +1,47 @@
-#include <Windows.h>
+#include <filesystem>
```

This function coverts an item to placeholder. It can be either a file or a folder. First we check if  it's a folder, then we create a handle of the item and using `winrt::file_handle` this handle is automatically closed when the function finishes so we don't have to manually do `CloseHandle`. After that we just call the Cloud Files API function `CfConvertToPlaceholder` with the item handler, the placeholderId and some other things that were already there before. We ignore the exception of `this item was already a placeholder` and we throw in case we have another error with `winrt::check_hresult`.